src/instructlab/model/evaluate.py

```diff
@@ -1,52 +1,145 @@
+
+
+def get_evaluator(
+    model_name,
+    benchmark,
+    judge_model_name,
+    output_dir,
+    max_workers,
+    taxonomy_path,
+    branch,
+    tasks,
+    few_shots,
+    batch_size,
+    task,
+    sdg_path,
+) -> Evaluator:
+    """takes in arguments from the CLI and uses 'benchmark' to validate other arguments
+    if all needed configuration is present, returns the appropriate Evaluator class for the benchmark
+    otherwise raises an exception for the missing values
+    """
+
+    # ensure skills benchmarks have proper arguments if selected
+    if benchmark in ["mt_bench", "mt_bench_branch"]:
+        required_args = [
+            model_name,
+            judge_model_name,
+            output_dir,
+            max_workers,
+        ]
+        if benchmark == "mt_bench_branch":
+            required_args.append(taxonomy_path, branch)
+        if any(required_args) is None:
+            click.secho(
+                f"Benchmark {benchmark} requires the following args to be set: {required_args}",
+                fg="red",
+            )
+            raise click.exceptions.Exit(1)
+        else:
+            evaluator_class = BENCHMARK_TO_CLASS_MAP[benchmark]
+            if benchmark == "mt_bench":
+                return evaluator_class(
+                    model_name, judge_model_name, output_dir, max_workers
+                )
+            else:
+                return evaluator_class(
+                    model_name,
+                    judge_model_name,
+                    taxonomy_path,
+                )
+
+    # ensure knowledge benchmarks have proper arguments if selected
+    if benchmark in ["mmlu", "mmlu_branch"]:
+        required_args = [few_shots, batch_size]
+        if benchmark == "mmlu":
+            required_args.append(tasks)
+        if benchmark == "mmlu_branch":
+            required_args.extend([task, sdg_path])
+        if any(required_args) is None:
+            click.secho(
+                f"Benchmark {benchmark} requires the following args to be set: {required_args}",
+                fg="red",
+            )
+            raise click.exceptions.Exit(1)
+        else:
+            evaluator_class = BENCHMARK_TO_CLASS_MAP[benchmark]
+            if benchmark == "mmlu":
+                return evaluator_class(tasks, few_shots, batch_size)
+            else:
+                return evaluator_class(sdg_path, task, few_shots, batch_size)
 
 
 @click.command(cls=DYMGroup)
+@click.option(
+    "--model-name",
+    type=click.Path(),
+    help="Path to model to be evaluated",
+)
 @click.option(
     "--benchmark",
-    type=click.Choice(list(benchmark_names_to_classes.keys())),
+    type=click.Choice(list(BENCHMARK_TO_CLASS_MAP.keys())),
     case_sensitive=False,
     help="Benchmarks to run during evaluation",
 )
 @click.option(
-    "--server-url",
+    "--judge-model-name",
+    type=click.Path(),
+    help="Path to model to be used as a judge for running mt_bench or mt_bench_branch",
+)
+@click.option(
```

Comment:
Should this have a default?

Reply:
I don't know.  Consistency with other cli dirs would be good.  The other consideration is whether we want to advertise the output to users or not.  Currently there isn't a reason to point users to it so I would suggest a . directory.

Reply:
Spoke with @cdoern, adding a default output dir called `eval_results` :+1: